src/core/common/timer.hpp

```diff
@@ -62,56 +64,70 @@ class Timer;
  */
 
 /**
- * This class implements the timer scheduler.
+ * The Alarm APIs definition
  *
  */
-class TimerScheduler
+struct AlarmApi
```

Comment:
Suggestion: We can consider adding the `struct AlarmApi` type definition in the `TimerSchedulerBase` class (as a `protected` type). It hides the definition and made it only accessible by `TimerSchedulerBase` and its subclasses.